src/java.base/share/classes/java/lang/template/SimpleStringTemplate.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.lang.template;
+
+import java.util.List;
+
+/**
+ * Generic StringTemplate.
+ *
+ * @param fragments  immutable list of string fragments from string template
+ * @param values     immutable list of expression values
+ */
+record SimpleStringTemplate(List<String> fragments,
+                             List<Object> values
+) implements StringTemplate {}
```

Comment:
A compact constructor doing defensive copies is missing

Reply:
The defensive copies are done by the callers.

Reply:
In that case, i wonder if not not better to move that record inside another class, closer to where the callers are

Reply:
Moving to TemplateRuntime